drivers/tty/serdev/pibridge.c

```diff
@@ -0,0 +1,453 @@
+#include <linux/errno.h>
+#include <linux/kfifo.h>
+#include <linux/module.h>
+#include <linux/mutex.h>
+#include <linux/of.h>
+#include <linux/serdev.h>
+#include <linux/pibridge_comm.h>
+#include <linux/wait.h>
+
+#define PIBRIDGE_BAUDRATE		115200
+#define PIBRIDGE_IO_TIMEOUT		10         // msec
+#define PIBRIDGE_RECV_BUFFER_SIZE	100
+#define PIBRIDGE_BC_ADDR		0xff
+
+#define PIBRIDGE_RESP_CMD		0x3fff
+#define PIBRIDGE_RESP_OK		0x4000
+#define PIBRIDGE_RESP_ERR		0x8000
+
+struct pibridge {
+	struct serdev_device *serdev;
+
+	struct mutex lock;
+	struct kfifo read_fifo;
+	wait_queue_head_t read_queue;
+};
+
+struct pibridge_pkthdr_gate {
+	u8	dst;
+	u8	src;
+	u16	cmd;
+	u16	seq;
+	u8	len;
+} __attribute__((packed));
+
+struct pibridge_pkthdr_io {
+	u8 addr	:6;
+	u8 type	:1;	/* 0 for unicast, 1 for broadcast */
+	u8 rsp	:1;	/*always be 0 for sending, might be 1 for receiving*/
```

Comment:
Looks like a missing space before and after the comment

Reply:
"might be" is a little confusing, it sounds like "perhaps it's 1 for responses but maybe it's not". I'd suggest something along the lines of: `/* 0 for request, 1 for response */`